File: MortageCalculator.java
import java.util.Scanner;

  public class MortageCalculator {

    public static void main(String [] args) {

      Scanner input = new Scanner(System.in);

      System.out.println("Enter principal: ");
      int principal = input.nextInt();

     System.out.println("Enter the annual intrest rate: ");
      double intrestRate = input.nextDouble();

    System.out.println("Enter the duration in years:");
     double duration = input.nextDouble();

    double numOfYears = 12 * duration;
    double rate = intrestRate/100;
    double monthlyRate = rate/12;
    double monthBase = 1 + monthlyRate;
    double base = Math.pow(monthBase , numOfYears);
    double number1 = (principal * monthlyRate);
    double number2 = (base - 1);
    double monthlyPayment = number1 * base / number2;

    System.out.printf("The monthly payment is %.2f " , monthlyPayment);
      
}
}

File: mortagecalculator.py
principal = int(input("Enter principal:"))

intrestrate = int(input("Enter annual intrest rate:"))

duration = int(input("Enter duration of years:"))

MONTH_IN_A_YEAR = 12

PERCENTAGE_RATE = 100

numofyears = MONTH_IN_A_YEAR * duration

rate = intrestrate / PERCENTAGE_RATE

monthlyrate = rate / MONTH_IN_A_YEAR

monthbase = 1 + monthlyrate

model = monthbase ** numofyears

number1 = principal * monthlyrate

number2 = model - 1

monthlypayment = number1 * model / number2

print(f" The monthly payment is : ${monthlypayment:,.2f}")